packages/modal-core/src/utils/CoreUtil.ts

```diff
@@ -36,10 +36,23 @@ export const CoreUtil = {
     return Array.isArray(data) && data.length > 0
   },
 
+  isTelegram() {
+    return (
+      (typeof window !== 'undefined' &&
+        // eslint-disable-next-line @typescript-eslint/no-explicit-any
+        Boolean((window as any).TelegramWebviewProxy)) ||
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
+      Boolean((window as any).Telegram) ||
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
+      Boolean((window as any).TelegramWebviewProxyProto)
+    )
+  },
```

Comment:
It'll break on other conditions, we should make a check on window before each individual checks
```suggestion
  isTelegram() {
    return typeof window !== 'undefined' && (
      // eslint-disable-next-line @typescript-eslint/no-explicit-any
      Boolean((window as any).TelegramWebviewProxy) ||
      // eslint-disable-next-line @typescript-eslint/no-explicit-any
      Boolean((window as any).Telegram) ||
      // eslint-disable-next-line @typescript-eslint/no-explicit-any
      Boolean((window as any).TelegramWebviewProxyProto
    )
  },
```